metadata-ingestion/src/datahub/ingestion/source/powerbi_report_server/report_server.py

```diff
@@ -541,6 +544,9 @@ def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
         reports = self.powerbi_client.get_all_reports()
 
         for report in reports:
+            if not self.source_config.report_pattern.allowed(report.id):
```

Comment:
`report.id` is uuid-like field, should we apply the pattern in something more _pattern_able such as the `report.name` instead?